### PR TITLE
fix: Danger contributors check

### PR DESCRIPTION
**Pull request by rogeruiz**
# Summary

This patch series adds better checking for Danger verifying if contributors was modified by only checking the JSON Diff. If there's not contributor object, then there is no change and move on. This eliminates all the false positives we've seen in recent PRs around modifying the package.json file near the word contributors.

## How To Test

You can create a new branch locally and modify the package.json with a contributors stanza. Then commit it and run the following Danger command to trigger it.

```
yarn danger local -b main --failOnErrors
```

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -126,43 +128,28 @@ const checkDependencyChanges: () => void = () => {
...
-      })
+// Check for any changes to the contributors section of package.json
+schedule(async () => {
+  if (!shouldRun) {
+    return;
   }
-}
+  const pd = await danger.git.JSONDiffForFile('package.json')
+
+  if (pd.contributors) {
```
Comment (rogeruiz) [inline dangerfile.ts:138]: If contributors exists in the diff, then we fail. This is a better check than seeing if contributors exists in the text diff.

File: dangerfile.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import * as child from 'child_process'
 
-import { danger, fail, warn } from 'danger'
+import { danger, fail, schedule, warn } from 'danger'
+
+const shouldRun = !danger.github || (danger.github && danger.github.pr.user.type !== 'Bot');
```
Comment (rogeruiz) [inline dangerfile.ts:5]: Add this up here as we're checking if we should run asynchronously now as well as functionally.

File: dangerfile.ts [hunk 3]
```diff
@@ -43,7 +45,7 @@ const checkYarnAudit: () => void = () => {
       })
       fail(
         `${issuesFound}${summary.data.vulnerabilities.high} high vulnerabilities and ` +
-          `${summary.data.vulnerabilities.critical} critical vulnerabilities found`
+        `${summary.data.vulnerabilities.critical} critical vulnerabilities found`
```
Comment (rogeruiz) [inline dangerfile.ts:48]: This was a formatting fix based setup.

**Review by rogeruiz — COMMENTED** [review]
This comments should help with the review.

**Review by brandonlenz — APPROVED** [review]

**Pull request merged**